todo/src/containers/AddTodo.js

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { connect } from "react-redux";
+import { addTodo } from "../components/actions";
+
+let AddTodo = ({ dispatch }) => {
+	let input;
+
+	return (
+		<div>
+			<form
+				onSubmit={e => {
+					e.preventDefault();
+					if (!input.value.trim()) {
+						return;
+					}
+					dispatch(addTodo(input.value));
```

Comment:
Rather than use dispatch, you should pass your action into your connect function to call it here in order to follow the redux pattern we are following in class.
```suggestion
					(addTodo(input.value));
```